packages/data-broker/src/services/dhis/DhisService.js

```diff
@@ -227,11 +227,12 @@ export class DhisService extends Service {
   };
 
   pullAnalyticsFromEventsForApi = async (api, dataSources, options) => {
-    const { programCodes = [], period, startDate, endDate } = options;
+    const { programCodes = [], period, startDate, endDate, organisationUnitCodes } = options;
     const dataElementCodes = dataSources.map(({ code }) => code);
     const dhisElementCodes = dataSources.map(({ dataElementCode }) => dataElementCode);
 
     const query = {
+      organisationUnitCodes,
```

Comment:
I'm not too sure...

Reply:
As per @kael89's comments on the other PR containing this change, could you please investigate this a bit before we approve this?

Reply:
Why it has worked before: https://github.com/beyondessential/tupaia/pull/1983#discussion_r575935785

However, we will not go ahead with this change:

The only difference between a getAnalytics call adding a dimension=ou:xxxx or not in an /api/analytics/events/query/<programId> call is that if you have the dimension=ou:xxxx param in the request, it returns the organisation id against each event in the response, otherwise it doesn't. 

Previously this was not needed since every other project works with org unit codes (which are present in both responses, with or without the ou param). Laos eoc is the first project to work with ids exclusively so it needs the change to add the ou param.

However, this change isn't going to work. With the addition of the ou param, it now sends 1700 entity ids in the DHIS request, and the server sends back a 400 Request URI too large. So that isn't going to work. 

We DO need to send an ou, but it can't be the facility. It could be the country.

At the very least, we should make the addition of the organisationUnitCodes change opt-in. DhisInputSchemeResolvingProxy can add it in IF it notices we are working with ids instead of codes, otherwise it can leave it out.

Reply:
That's some great investigation @IgorNadj ! So `ou` is optional in `/api/analytics/events/query/`? If that's the case, how can we be sure that we retrieve data for the selected org unit (and not for others) the front end?

Is it a coincidence (eg always fetching country level data) that this currently works, or do we filter the fetched events in some other way?

Reply:
Created https://github.com/beyondessential/tupaia-backlog/issues/2251 to investigate

Reply:
Another option for resolving the request too large issue is fetching in batches.

![image](https://user-images.githubusercontent.com/1274422/108001919-ac94e380-7052-11eb-96a3-56e7f2828e71.png)

It does come with a bit of a performance hit

Reply:
> 🤔 how has this even been working....?

Investigated and fixed in https://github.com/beyondessential/tupaia-backlog/issues/2251

The remaining issue of batching the ou param still needs to be fixed. Love the idea @edmofro!

Reply:
Moved out the batching to another comment, resolving this thread